packages/web/src/features/translation/TranslationView.tsx

```diff
@@ -174,6 +178,24 @@ export default function TranslationView() {
     },
   });
 
+  const [translationName, setTranslationName] = useState('');
+  const [verseTranslation, setVerseTranslation] = useState('');
+  useEffect(() => {
+    // Clear the current translation content while loading.
+    setVerseTranslation('');
+    bibleTranslationClient
```

Comment:
We should use `useQuery` instead of `useEffect` since we are dealing with the network. If you don't use something like`useEffect` scenarios like this can happen:

1. Navigate to verse 1
2. Start loading translations for verse 1 - let's say this takes a long time to load due to network issues
3. Navigate to verse 2
4. Start loading translations for verse 2 - this one loads quickly because the network issues are resolved
5. Verse 2 request completes
6. Verse 1 request completes

Since the verse 1 request completes second, it's callback executes second and we end up displaying the wrong verse.